### PR TITLE
MAUI-4756-Cell template and Actions Buttons UG in .NET MAUI

### PR DESCRIPTION
**Pull request by Priyadharshini-Periyasamy**
#### Added the UG content for the below Topics:

- Cell template
- Actions buttons
- Today button
- Special day predicate

### CODE REVIEW

File: MAUI/Calendar/customizations.md [hunk 1]
```diff
@@ -93,6 +91,79 @@ You can customize the calendar month view cell by using the `MonthView` property
 
 ![Month view Customization in .NET MAUI Calendar.](images/customization/maui-month-view-customization.png)
 
+*    **Special day predicate** - You can add special dates to the `Calendar` by using the [SpecialDates](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Calendar.CalendarMonthView.html#Syncfusion_Maui_Calendar_CalendarMonthView_SpecialDates) property and you can also customize the special dates text style and background of the `Calendar` by using the [SpecialDatesTextStyle](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Calendar.CalendarMonthView.html#Syncfusion_Maui_Calendar_CalendarMonthView_SpecialDatesTextStyle) and [SpecialDatesBackground](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Calendar.CalendarMonthView.html#Syncfusion_Maui_Calendar_CalendarMonthView_SpecialDatesBackground) properties of [MonthView](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Calendar.CalendarMonthView.html).
```
Comment (florenceliliana) [inline MAUI/Calendar/customizations.md:94]: * **Special day predicate** - You can add special dates to the `Calendar` by using the [SpecialDates](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Calendar.CalendarMonthView.html#Syncfusion_Maui_Calendar_CalendarMonthView_SpecialDates) property, and you can also customize the special dates text style and background of the `Calendar` by using the [SpecialDatesTextStyle](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Calendar.CalendarMonthView.html#Syncfusion_Maui_Calendar_CalendarMonthView_SpecialDatesTextStyle) and [SpecialDatesBackground](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Calendar.CalendarMonthView.html#Syncfusion_Maui_Calendar_CalendarMonthView_SpecialDatesBackground) properties of [MonthView](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Calendar.CalendarMonthView.html).

File: MAUI/Calendar/customizations.md [hunk 2]
```diff
@@ -93,6 +91,79 @@ You can customize the calendar month view cell by using the `MonthView` property
...
+
+Special day Predicate decides whether the month cell date is a special date or not in the calendar. It supports icon shapes in month view to display such as dots, hearts, diamonds, stars, and bells if special dates are specified. The icon colors are also customizable. We can improve the special dates support with this special day predicate by displaying icons.
```
Comment (florenceliliana) [inline MAUI/Calendar/customizations.md:96]: The special day Predicate decides whether the month cell date is a special date or not in the calendar. If special dates are specified, it supports the icon shapes in the month views display, such as dots, hearts, diamonds, stars, and bells. The icon colors are also customizable. By displaying icons, we can improve the special dates support with this special day predicate.

File: MAUI/Calendar/customizations.md [hunk 3]
```diff
@@ -222,4 +293,252 @@ You can customize the selection cell background and text style in Month, Year, D
 {% endhighlight %}
 {% endtabs %}
 
-![Month view Range Selection in .NET MAUI Calendar.](images/customization/maui-month-view-range-selection.png)
\ No newline at end of file
+![Month view Range Selection in .NET MAUI Calendar.](images/customization/maui-month-view-range-selection.png)
+
+## Month cell appearance using DataTemplate
+
+The month cell appearance can be customized by using the `CellTemplate` property of `MonthView` in the `SfCalendar`.
```
Comment (florenceliliana) [inline MAUI/Calendar/customizations.md:300]: The month cell appearance can be customized using the `CellTemplate` property of `MonthView` in the `SfCalendar.`

File: MAUI/Calendar/customizations.md [hunk 4]
```diff
@@ -222,4 +293,252 @@ You can customize the selection cell background and text style in Month, Year, D
...
+    grid.Children.Add(label);
+    return grid;
+});
+
+{% endhighlight %}
+{% endtabs %}
+
+## Month cell appearance using DataTemplateSelector
+
+You can customize the month cell appearance by using the `CellTemplate` property of `MonthView` in the `SfCalendar`. The `DataTemplateSelector` for a calendar can be used to choose a different data template for each cell in the calendar based on the cell's data. It allows to customize the appearance of a particular cell based on certain conditions.
```
Comment (florenceliliana) [inline MAUI/Calendar/customizations.md:355]: You can customize the month cell appearance using the `CellTemplate` property of `MonthView` in the `SfCalendar.` The `DataTemplateSelector` for a calendar can be used to choose a different data template for each cell in the calendar based on the cell's data. It allows customizing the appearance of a particular cell based on certain conditions.

File: MAUI/Calendar/customizations.md [hunk 5]
```diff
@@ -222,4 +293,252 @@ You can customize the selection cell background and text style in Month, Year, D
...
+{% highlight c# tabtitle="MainPage.xaml.cs" %}
+
+this.Calendar.View = CalendarView.Month;
+
+{% endhighlight %}
+{% endtabs %}
+
+## Year cell appearance using DataTemplate
+
+The year cell appearance can be customized by using the `CellTemplate` property of `YearView` in the `SfCalendar`.
```
Comment (florenceliliana) [inline MAUI/Calendar/customizations.md:423]: The year cell appearance can be customized using the `CellTemplate` property of `YearView` in the `SfCalendar.`

File: MAUI/Calendar/customizations.md [hunk 6]
```diff
@@ -222,4 +293,252 @@ You can customize the selection cell background and text style in Month, Year, D
...
+    grid.Children.Add(label);
+    return grid;
+});
+
+{% endhighlight %}
+{% endtabs %}
+
+## Year cell appearance using DataTemplateSelector
+
+You can customize the month cell appearance by using the `CellTemplate` property of `YearView` in the `SfCalendar`. The `DataTemplateSelector` for a calendar can be used to choose a different data template for each cell in the calendar based on the cell's data. It allows to customize the appearance of a particular cell based on certain conditions.
```
Comment (florenceliliana) [inline MAUI/Calendar/customizations.md:478]: You can customize the month cell appearance by using the `CellTemplate` property of `YearView` in the `SfCalendar.` The `DataTemplateSelector` for a calendar can be used to choose a different data template for each cell in the calendar based on the cell's data. It allows customizing the appearance of a particular cell based on certain conditions.

File: MAUI/Calendar/getting-started.md [hunk 7]
```diff
@@ -156,4 +156,34 @@ this.Calendar.SelectionMode = CalendarSelectionMode.Multiple;
 {% endhighlight %}  
 {% endtabs %}
 
-![change-selection-mode](images/getting-started/maui-calendar-multiple-selection.png)
\ No newline at end of file
+![change-selection-mode](images/getting-started/maui-calendar-multiple-selection.png)
+
+## Action buttons
+
+You can display action buttons at the bottom of the calendar by using the `ShowActionButtons` property of SfCalendar. It allows you to confirm or cancel the selected date, dates, or date ranges of the `SfCalendar`.
```
Comment (florenceliliana) [inline MAUI/Calendar/getting-started.md:163]: You can display action buttons at the bottom of the calendar by using the `ShowActionButtons` property of the SfCalendar. It allows you to confirm or cancel the selected date, dates, or date ranges of the `SfCalendar.`

File: MAUI/Calendar/getting-started.md [hunk 8]
```diff
@@ -156,4 +156,34 @@ this.Calendar.SelectionMode = CalendarSelectionMode.Multiple;
...
+
+## Today button
+
+The today button can be displayed at the bottom of the calendar by using the `ShowTodayButton` property of the `SfCalendar`, allowing you to switch from the current view to today's view effortlessly.
```
Comment (florenceliliana) [inline MAUI/Calendar/getting-started.md:167]: The today button can be displayed at the bottom of the calendar using the `ShowTodayButton` property of the `SfCalendar,` allowing you to switch from the current view to today's view effortlessly.

File: MAUI/Calendar/getting-started.md [hunk 9]
```diff
@@ -156,4 +156,34 @@ this.Calendar.SelectionMode = CalendarSelectionMode.Multiple;
...
+
+The following code explains how to enable the Action button and Today button,
```
Comment (florenceliliana) [inline MAUI/Calendar/getting-started.md:169]: The following code explains how to enable the Action button and Today button.

**Review by florenceliliana — CHANGES_REQUESTED** [review]

File: MAUI/Calendar/customizations.md [hunk 10]
```diff
@@ -93,6 +91,79 @@ You can customize the calendar month view cell by using the `MonthView` property
 
 ![Month view Customization in .NET MAUI Calendar.](images/customization/maui-month-view-customization.png)
 
+*    **Special day predicate** - You can add special dates to the `Calendar` by using the [SpecialDates](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Calendar.CalendarMonthView.html#Syncfusion_Maui_Calendar_CalendarMonthView_SpecialDates) property, and you can also customize the special dates text style and background of the `Calendar` by using the [SpecialDatesTextStyle](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Calendar.CalendarMonthView.html#Syncfusion_Maui_Calendar_CalendarMonthView_SpecialDatesTextStyle) and [SpecialDatesBackground](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Calendar.CalendarMonthView.html#Syncfusion_Maui_Calendar_CalendarMonthView_SpecialDatesBackground) properties of [MonthView](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Calendar.CalendarMonthView.html).
```
Comment (Sasikumar3595) [inline MAUI/Calendar/customizations.md:94]: SpecialDates - Property is deprecated

File: MAUI/Calendar/customizations.md [hunk 11]
```diff
@@ -93,6 +91,79 @@ You can customize the calendar month view cell by using the `MonthView` property
...
+
+The special day Predicate decides whether the month cell date is a special date or not in the calendar. If special dates are specified, it supports the icon shapes in the month views to display, such as dots, hearts, diamonds, stars, and bells. The icon colors are also customizable. By displaying icons, we can improve the special dates support with this special day predicate.
```
Comment (Sasikumar3595) [inline MAUI/Calendar/customizations.md:96]: Improve the content

File: MAUI/Calendar/getting-started.md [hunk 12]
```diff
@@ -156,4 +156,34 @@ this.Calendar.SelectionMode = CalendarSelectionMode.Multiple;
 {% endhighlight %}  
 {% endtabs %}
 
-![change-selection-mode](images/getting-started/maui-calendar-multiple-selection.png)
\ No newline at end of file
+![change-selection-mode](images/getting-started/maui-calendar-multiple-selection.png)
+
+## Action buttons
+
+You can display action buttons at the bottom of the calendar by using the `ShowActionButtons` property of the SfCalendar. It allows you to confirm or cancel the selected date, dates, or date ranges of the `SfCalendar`.
```
Comment (Sasikumar3595) [inline MAUI/Calendar/getting-started.md:163]: Add code for action button and today seperate

File: MAUI/Calendar/customizations.md [hunk 13]
```diff
@@ -222,4 +293,346 @@ You can customize the selection cell background and text style in Month, Year, D
...
+}
+
+{% highlight c# tabtitle="MainPage.xaml.cs" %}
+
+this.Calendar.View = CalendarView.Decade;
+
+{% endhighlight %}
+{% endtabs %}
+
+![Cell temaplte in .NET MAUI Calendar.](images/customization/maui-cell_template.png)
```
Comment (Sasikumar3595) [inline MAUI/Calendar/customizations.md:638]: This images are not relatable for above code, remove the images

File: MAUI/Calendar/migration.md [hunk 14]
```diff
@@ -18,8 +18,8 @@ To make the migration from the [Xamarin SfCalendar](https://www.syncfusion.com/x
 <th>Xamarin SfCalendar</th>
 <th>.NET MAUI SfCalendar</th></tr>
 <tr>
-<td>Syncfusion.SfCalendar.XForms</td>
-<td>Syncfusion.Maui.SfCalendar</td></tr>
+<td>Syncfusion.Calendar.XForms</td>
```
Comment (Sasikumar3595) [inline MAUI/Calendar/migration.md:21]: Syncfusion.SfCalendar.XForms - Check the namespace

File: MAUI/Calendar/customizations.md (hunk 10 above)
Comment (Priyadharshini-Periyasamy) [reply MAUI/Calendar/customizations.md]: Special dates property removed and added the content for special day predicate.

File: MAUI/Calendar/customizations.md (hunk 11 above)
Comment (Priyadharshini-Periyasamy) [reply MAUI/Calendar/customizations.md]: Improved the special day predicate content

File: MAUI/Calendar/customizations.md [hunk 15]
```diff
@@ -93,6 +91,79 @@ You can customize the calendar month view cell by using the `MonthView` property
...
+
+<calendar:SfCalendar  x:Name="Calendar"  View="Month">
+</calendar:SfCalendar>
+
+{% endhighlight %}
+{% highlight c# tabtitle="MainPage.xaml.cs" %}
+
+this.Calendar.MonthView.SpecialDayPredicate = (date) =>
+{
+    if (date.Date == DateTime.Now.AddDays(2).Date)
```
[the remark this answers is not in the document]
Comment (Priyadharshini-Periyasamy) [reply MAUI/Calendar/customizations.md:107]: Checked the shapes and it is right based on the image.

File: MAUI/Calendar/customizations.md (hunk 13 above)
Comment (Priyadharshini-Periyasamy) [reply MAUI/Calendar/customizations.md]: The mentioned images are removed.

File: MAUI/Calendar/getting-started.md (hunk 12 above)
Comment (Priyadharshini-Periyasamy) [reply MAUI/Calendar/getting-started.md]: Added the example code for action and today button seperately.

File: MAUI/Calendar/migration.md (hunk 14 above)
Comment (Priyadharshini-Periyasamy) [reply MAUI/Calendar/migration.md]: Checked and changed the namespace to Syncfusion.SfCalendar.XForms

File: MAUI/Calendar/customizations.md [hunk 16]
```diff
@@ -93,6 +91,77 @@ You can customize the calendar month view cell by using the `MonthView` property
 
 ![Month view Customization in .NET MAUI Calendar.](images/customization/maui-month-view-customization.png)
 
+*    **Special day predicate** - The special day Predicate decides whether the month cell date is a special date or not in the calendar.You can add special date to the `Calendar` by using the `SpecialDayPredicate` property, and you can also customize the special day text style and background of the `Calendar` by using the [SpecialDatesTextStyle](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Calendar.CalendarMonthView.html#Syncfusion_Maui_Calendar_CalendarMonthView_SpecialDatesTextStyle) and [SpecialDatesBackground](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Calendar.CalendarMonthView.html#Syncfusion_Maui_Calendar_CalendarMonthView_SpecialDatesBackground) properties of [MonthView](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Calendar.CalendarMonthView.html). You can customize the special day with different icons such as dots, hearts, diamonds, stars, and bells in the `MonthView` and also customize the icon color. If you do not wish to add any icon details, you can set them to be transparent.
```
Comment (Sasikumar3595) [inline MAUI/Calendar/customizations.md:94]: The special day Predicate 

Update the name as predicate

Comment (Sasikumar3595) [inline MAUI/Calendar/customizations.md:94]: Prepare the content review task and get content review

File: MAUI/Calendar/customizations.md (hunk 1 above)
Comment (Priyadharshini-Periyasamy) [reply MAUI/Calendar/customizations.md]: Content modified.

File: MAUI/Calendar/customizations.md (hunk 2 above)
Comment (Priyadharshini-Periyasamy) [reply MAUI/Calendar/customizations.md]: Content modified.

File: MAUI/Calendar/customizations.md (hunk 3 above)
Comment (Priyadharshini-Periyasamy) [reply MAUI/Calendar/customizations.md]: Content modified.

File: MAUI/Calendar/customizations.md (hunk 4 above)
Comment (Priyadharshini-Periyasamy) [reply MAUI/Calendar/customizations.md]: Content modified.

File: MAUI/Calendar/customizations.md (hunk 5 above)
Comment (Priyadharshini-Periyasamy) [reply MAUI/Calendar/customizations.md]: Content modified.

File: MAUI/Calendar/customizations.md (hunk 6 above)
Comment (Priyadharshini-Periyasamy) [reply MAUI/Calendar/customizations.md]: Content modified.

File: MAUI/Calendar/getting-started.md (hunk 7 above)
Comment (Priyadharshini-Periyasamy) [reply MAUI/Calendar/getting-started.md]: Content modified.

File: MAUI/Calendar/getting-started.md (hunk 8 above)
Comment (Priyadharshini-Periyasamy) [reply MAUI/Calendar/getting-started.md]: Content modified.

File: MAUI/Calendar/getting-started.md (hunk 9 above)
Comment (Priyadharshini-Periyasamy) [reply MAUI/Calendar/getting-started.md]: Content modified.

File: MAUI/Calendar/customizations.md [hunk 17]
```diff
@@ -17,13 +17,11 @@ You can customize the calendar month view cell by using the `MonthView` property
...
+*    **Trailing and leading dates** – You can hide the trailing and leading dates by using the [ShowTrailingAndLeadingDates](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Calendar.SfCalendar.html#Syncfusion_Maui_Calendar_SfCalendar_ShowTrailingAndLeadingDates) property in the [CalendarMonthView](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Calendar.CalendarMonthView.html) class. You can also customize the trailing and leading dates text style and background of the `Calendar` by using the [TrailingLeadingDatesTextStyle](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Calendar.CalendarMonthView.html#Syncfusion_Maui_Calendar_CalendarMonthView_TrailingLeadingDatesTextStyle) and [TrailingLeadingDatesBackground](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Calendar.CalendarMonthView.html#Syncfusion_Maui_Calendar_CalendarMonthView_TrailingLeadingDatesBackground) properties of [MonthView](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Calendar.CalendarMonthView.html).
```
Comment (florenceliliana) [inline MAUI/Calendar/customizations.md:20]: * **Trailing and leading dates** – You can hide the trailing and leading dates using the [ShowTrailingAndLeadingDates](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Calendar.SfCalendar.html#Syncfusion_Maui_Calendar_SfCalendar_ShowTrailingAndLeadingDates) property in the [CalendarMonthView](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Calendar.CalendarMonthView.html) class. You can also customize the trailing and leading dates text style and background of the `Calendar` using the [TrailingLeadingDatesTextStyle](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Calendar.CalendarMonthView.html#Syncfusion_Maui_Calendar_CalendarMonthView_TrailingLeadingDatesTextStyle) and [TrailingLeadingDatesBackground](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Calendar.CalendarMonthView.html#Syncfusion_Maui_Calendar_CalendarMonthView_TrailingLeadingDatesBackground) properties of the [MonthView](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Calendar.CalendarMonthView.html).

File: MAUI/Calendar/customizations.md [hunk 18]
```diff
@@ -17,13 +17,11 @@ You can customize the calendar month view cell by using the `MonthView` property
...
-
-*    **Weekend Dates** – You can customize the weekend dates text style and background of the `Calendar` by using the [WeekendDatesTextStyle](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Calendar.CalendarMonthView.html#Syncfusion_Maui_Calendar_CalendarMonthView_WeekendDatesTextStyle) and [WeekendDatesBackground](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Calendar.CalendarMonthView.html#Syncfusion_Maui_Calendar_CalendarMonthView_WeekendDatesBackground) properties of [MonthView](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Calendar.CalendarMonthView.html).
+*    **Weekend dates** – You can customize the weekend dates text style and background of the `Calendar` by using the [WeekendDatesTextStyle](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Calendar.CalendarMonthView.html#Syncfusion_Maui_Calendar_CalendarMonthView_WeekendDatesTextStyle) and [WeekendDatesBackground](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Calendar.CalendarMonthView.html#Syncfusion_Maui_Calendar_CalendarMonthView_WeekendDatesBackground) properties of [MonthView](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Calendar.CalendarMonthView.html).
```
Comment (florenceliliana) [inline MAUI/Calendar/customizations.md:24]: * **Weekend dates** – You can customize the weekend dates text style and background of the `Calendar` using the [WeekendDatesTextStyle](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Calendar.CalendarMonthView.html#Syncfusion_Maui_Calendar_CalendarMonthView_WeekendDatesTextStyle) and [WeekendDatesBackground](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Calendar.CalendarMonthView.html#Syncfusion_Maui_Calendar_CalendarMonthView_WeekendDatesBackground) properties of the [MonthView](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Calendar.CalendarMonthView.html).

File: MAUI/Calendar/customizations.md [hunk 19]
```diff
@@ -93,6 +91,77 @@ You can customize the calendar month view cell by using the `MonthView` property
 
 ![Month view Customization in .NET MAUI Calendar.](images/customization/maui-month-view-customization.png)
 
+*    **Special day predicate** - The special day predicate decides whether the month cell date is a special date or not in the calendar.You can add special date to the `Calendar` by using the `SpecialDayPredicate` property, and you can also customize the special day text style and background of the `Calendar` by using the [SpecialDatesTextStyle](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Calendar.CalendarMonthView.html#Syncfusion_Maui_Calendar_CalendarMonthView_SpecialDatesTextStyle) and [SpecialDatesBackground](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Calendar.CalendarMonthView.html#Syncfusion_Maui_Calendar_CalendarMonthView_SpecialDatesBackground) properties of [MonthView](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Calendar.CalendarMonthView.html). You can customize the special day with different icons such as dots, hearts, diamonds, stars, and bells in the `MonthView` and also customize the icon color. If you do not wish to add any icon details, you can set them to be transparent.
```
Comment (florenceliliana) [inline MAUI/Calendar/customizations.md:94]: * **Special day predicate** - The special day predicate decides whether the month cell date is a special date or not in the calendar. You can add a special date to the `Calendar` using the `SpecialDayPredicate` property, and you can also customize the special day text style and background of the `Calendar` using the [SpecialDatesTextStyle](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Calendar.CalendarMonthView.html#Syncfusion_Maui_Calendar_CalendarMonthView_SpecialDatesTextStyle) and [SpecialDatesBackground](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Calendar.CalendarMonthView.html#Syncfusion_Maui_Calendar_CalendarMonthView_SpecialDatesBackground) properties of [MonthView](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Calendar.CalendarMonthView.html). You can customize the special day with icons such as dots, hearts, diamonds, stars, and bells in the `MonthView` and also customize the icon color. If you do not wish to add any icon details, you can set them to be transparent.

File: MAUI/Calendar/customizations.md [hunk 20]
```diff
@@ -222,4 +291,352 @@ You can customize the selection cell background and text style in Month, Year, D
 {% endhighlight %}
 {% endtabs %}
 
-![Month view Range Selection in .NET MAUI Calendar.](images/customization/maui-month-view-range-selection.png)
\ No newline at end of file
+![Month view Range Selection in .NET MAUI Calendar.](images/customization/maui-month-view-range-selection.png)
+
+## Month cell appearance using DataTemplate
+
+The month cell appearance can be customized using the `CellTemplate` property of `MonthView` in the `SfCalendar`.
```
Comment (florenceliliana) [inline MAUI/Calendar/customizations.md:298]: The month cell appearance can be customized using the `CellTemplate` property of `MonthView` in the `SfCalendar.`

File: MAUI/Calendar/customizations.md [hunk 21]
```diff
@@ -222,4 +291,352 @@ You can customize the selection cell background and text style in Month, Year, D
...
+        return null;
+    }
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+## Month cell appearance using DataTemplateSelector
+
+You can customize the month cell appearance using the `CellTemplate` property of `MonthView` in the `SfCalendar`. The `DataTemplateSelector` for a calendar can be used to choose a different data template for each cell in the calendar based on the cell's data. It allows customizing the appearance of a particular cell based on certain conditions.
```
Comment (florenceliliana) [inline MAUI/Calendar/customizations.md:405]: You can customize the month cell appearance using the `CellTemplate` property of the `MonthView` in the `SfCalendar.` The `DataTemplateSelector` for a calendar can be used to choose a different data template for each cell in the calendar based on the cell's data. It allows customizing the appearance of a particular cell based on certain conditions.

File: MAUI/Calendar/customizations.md [hunk 22]
```diff
@@ -222,4 +291,352 @@ You can customize the selection cell background and text style in Month, Year, D
...
+{% highlight c# tabtitle="MainPage.xaml.cs" %}
+
+this.Calendar.View = CalendarView.Month;
+
+{% endhighlight %}
+{% endtabs %}
+
+## Year cell appearance using DataTemplate
+
+The year cell appearance can be customized using the `CellTemplate` property of `YearView` in the `SfCalendar`.
```
Comment (florenceliliana) [inline MAUI/Calendar/customizations.md:473]: The year cell appearance can be customized using the `CellTemplate` property of `YearView` in the `SfCalendar.`

File: MAUI/Calendar/customizations.md [hunk 23]
```diff
@@ -222,4 +291,352 @@ You can customize the selection cell background and text style in Month, Year, D
...
+        return null;
+    }
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+## Year cell appearance using DataTemplateSelector
+
+You can customize the month cell appearance by using the `CellTemplate` property of `YearView` in the `SfCalendar`. The `DataTemplateSelector` for a calendar can be used to choose a different data template for each cell in the calendar based on the cell's data. It allows customizing the appearance of a particular cell based on certain conditions.
```
Comment (florenceliliana) [inline MAUI/Calendar/customizations.md:578]: You can customize the month cell appearance by using the `CellTemplate` property of `YearView` in the `SfCalendar.` The `DataTemplateSelector` for a calendar can be used to choose a different data template for each cell in the calendar based on the cell's data. It allows customizing the appearance of a particular cell based on certain conditions.

File: MAUI/Calendar/getting-started.md [hunk 24]
```diff
@@ -156,4 +156,44 @@ this.Calendar.SelectionMode = CalendarSelectionMode.Multiple;
 {% endhighlight %}  
 {% endtabs %}
 
-![change-selection-mode](images/getting-started/maui-calendar-multiple-selection.png)
\ No newline at end of file
+![change-selection-mode](images/getting-started/maui-calendar-multiple-selection.png)
+
+## Action buttons
+
+You can display action buttons at the bottom of the calendar by using the `ShowActionButtons` property of the SfCalendar. It allows you to confirm or cancel the selected date, dates, and range of dates in calendar views of the `SfCalendar`.
```
Comment (florenceliliana) [inline MAUI/Calendar/getting-started.md:163]: You can display action buttons at the bottom of the calendar by using the `ShowActionButtons` property of the SfCalendar. It allows you to confirm or cancel the selected date, dates, and range of dates in calendar views of the `SfCalendar.`

File: MAUI/Calendar/getting-started.md [hunk 25]
```diff
@@ -156,4 +156,44 @@ this.Calendar.SelectionMode = CalendarSelectionMode.Multiple;
...
+{% highlight c# tabtitle="C#" %}
+
+this.Calendar.ShowActionButtons = true;
+
+{% endhighlight %}  
+{% endtabs %}
+
+## Today button
+
+The today button can be displayed at the bottom of the calendar using the `ShowTodayButton` property of the `SfCalendar`, allowing you to quickly navigate from current view to the today view.
```
Comment (florenceliliana) [inline MAUI/Calendar/getting-started.md:183]: The today button can be displayed at the bottom of the calendar using the `ShowTodayButton` property of the `SfCalendar,` allowing you to quickly navigate from the current view to the today view.

File: MAUI/Calendar/customizations.md (hunk 16 above)
Comment (Priyadharshini-Periyasamy) [reply MAUI/Calendar/customizations.md]: The special day Predicate updated as predicate.

Comment (Priyadharshini-Periyasamy) [reply MAUI/Calendar/customizations.md]: Prepared the content review task and content reviewed

File: MAUI/Calendar/customizations.md [hunk 26]
```diff
@@ -222,4 +291,344 @@ You can customize the selection cell background and text style in Month, Year, D
...
+            grid.Padding = new Thickness(1);
+
+            return grid;
+        });
+
+        this.template = this.circleTemplate;
+    }
+}
+internal class DateToStrokeConverter : IValueConverter
+{
```
[the remark this answers is not in the document]
Comment (Priyadharshini-Periyasamy) [reply MAUI/Calendar/customizations.md:382]: Added the new page as Value converter.cs

File: MAUI/Calendar/customizations.md [hunk 27]
```diff
@@ -222,4 +291,344 @@ You can customize the selection cell background and text style in Month, Year, D
...
+        <calendar:SfCalendar.MonthView>
+            <calendar:CalendarMonthView CellTemplate="{StaticResource monthCellTemplateSelector}" />
+        </calendar:SfCalendar.MonthView>
+    </calendar:SfCalendar>
+    </Grid>
+
+{% endhighlight %}
+{% highlight c# tabtitle="MonthCellTemplateSelector.cs" %}
+
+public class MonthCellTemplateSelector : DataTemplateSelector
```
[the remark this answers is not in the document]
Comment (Priyadharshini-Periyasamy) [reply MAUI/Calendar/customizations.md:443]: Added the title as Template selector.cs

File: MAUI/Calendar/customizations.md (hunk 17 above)
Comment (Priyadharshini-Periyasamy) [reply MAUI/Calendar/customizations.md]: Contents modified.

File: MAUI/Calendar/customizations.md (hunk 19 above)
Comment (Priyadharshini-Periyasamy) [reply MAUI/Calendar/customizations.md]: Content modified.

File: MAUI/Calendar/customizations.md (hunk 20 above)
Comment (Priyadharshini-Periyasamy) [reply MAUI/Calendar/customizations.md:298]: Contents modified.

File: MAUI/Calendar/customizations.md (hunk 21 above)
Comment (Priyadharshini-Periyasamy) [reply MAUI/Calendar/customizations.md:405]: Contents modified.

File: MAUI/Calendar/customizations.md (hunk 23 above)
Comment (Priyadharshini-Periyasamy) [reply MAUI/Calendar/customizations.md]: Content modified.

File: MAUI/Calendar/getting-started.md (hunk 24 above)
Comment (Priyadharshini-Periyasamy) [reply MAUI/Calendar/getting-started.md:163]: Content modified.

File: MAUI/Calendar/getting-started.md (hunk 25 above)
Comment (Priyadharshini-Periyasamy) [reply MAUI/Calendar/getting-started.md:183]: Content modified.

**Review by Nijamudeen-Mohamed — APPROVED** [review]

**Pull request merged**